test/ex_doc/autolink_test.exs

```diff
@@ -168,11 +175,37 @@ defmodule ExDoc.AutolinkTest do
       assert autolink(~m"[custom text](`:lists.all/2`)") ==
                ~m"[custom text](http://www.erlang.org/doc/man/lists.html#all-2)"
 
-      # TODO: with custom links and backticks there are no false positives (you
-      #       always mean to link) so we should always warn on mismatches?
-      #       Though backticks are markdown specific, is that ok?
-      # assert_warn(fn ->
-      assert_unchanged(~m"[custom text](`Unknown`)")
+      captured =
+        assert_warn(fn ->
+          assert_unchanged(~m"[custom text](`Unknown`)")
+        end)
+
+      assert captured =~
+               "documentation references module \"Unknown\" but it is undefined"
+
+      captured =
+        assert_warn(fn ->
+          assert_unchanged(~m"[custom text](Unknown)")
+        end)
+
+      assert captured =~
+               "documentation references file \"Unknown\" but it doesn't exist"
+
+      captured =
+        assert_warn(fn ->
+          assert_unchanged(~m"[custom text](`LICENSE`)", extras: ["LICENSE"])
+        end)
+
+      assert captured =~
+               "documentation references module \"LICENSE\" but it is undefined"
+
+      captured =
+        assert_warn(fn ->
+          assert_unchanged(~m"[an unknown task](`mix unknown.task`)")
+        end)
+
+      assert captured =~
+               "documentation references \"mix unknown.task\" but such task is undefined"
```

Comment:
could you also add the following test? `assert_unchanged(~m"[an unknown task](`Mix.Tasks.Unknown`)")`

the warning isn't ideal but that's ok.

Reply:
I think we need to store exact string that the user passed, and print it out in the error. Another example is if you call `Elixir.String.upcase/9`. It will be printed out as `String.upcase/9`.
I am submitting another PR that deals with the `Elixir.` namespace, but printing the original string has been left out, so this could potentially be included in the warn and maybe_warn refactoring

Reply:
Yes, I too think keeping the original is the way to go because it will aid in finding the broken ref in the docs. Good call.

Reply:
tackled in https://github.com/elixir-lang/ex_doc/pull/1235/commits/e0a7e493445f6946902e85facfceb4ac914002fa